pegasus/routes/poste_routes.rb

```diff
@@ -27,7 +31,9 @@
   only_for 'code.org'
   dont_cache
   delivery = DB[:poste_deliveries].where(id: Poste.decrypt_id(id)).first
-  id = DB[:poste_opens].insert(delivery_id: delivery[:id], created_ip: request.ip, created_at: DateTime.now) if delivery
+  pass unless delivery
+
+  id = DB[:poste_opens].insert(delivery_id: delivery[:id], created_ip: request.ip, created_at: DateTime.now)
```

Comment:
nit: For readability, I would do `s/id/poste_opens_id`, updating the `headers` appropriately.